### PR TITLE
Add a model mesh demo

### PR DESCRIPTION
**Pull request by gz65555**
—— load a standford bunny obj model.

### CODE REVIEW

**Review by singlecoder — APPROVED** [review]
+1

File: demos/model-mesh-obj-loader/index.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+engine.canvas.resizeByClientSize();
+const rootEntity = engine.sceneManager.activeScene.createRootEntity();
+
+// init camera
+const cameraEntity = rootEntity.createChild("camera");
+cameraEntity.addComponent(o3.Camera);
+cameraEntity.addComponent(OrbitControl);
+const pos = cameraEntity.transform.position;
+pos.setValue(0.5, 0.5, 0.5);
+cameraEntity.transform.position = pos;
```
Comment (GuoLei1990) [inline demos/model-mesh-obj-loader/index.ts:15]: cameraEntity.transform.setPosition(0.5, 0.5, 0.5)

File: demos/model-mesh-obj-loader/index.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import * as o3 from "oasis-engine";
```
Comment (GuoLei1990) [inline demos/model-mesh-obj-loader/index.ts:1]: use  import {
***,
***
} from "oasis-engine";

**Comment by GuoLei1990** [comment]
Your commit account is not github

File: demos/model-mesh-obj-loader/index.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        } else if (parseTexts[0] === "f") {
+          indices.push(parseInt(parseTexts[1]) - 1, parseInt(parseTexts[2]) - 1, parseInt(parseTexts[3]) - 1);
+        }
+      });
+
+    const mesh = new o3.ModelMesh(engine);
+    mesh.setPositions(positions);
+    mesh.setIndices(Uint16Array.from(indices));
+    mesh.addSubMesh(0, indices.length, MeshTopology.Triangles);
+    mesh.uploadData(false);
```
Comment (GuoLei1990) [inline demos/model-mesh-obj-loader/index.ts:44]: mesh.uploadData(true) is better

File: demos/model-mesh-obj-loader/index.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import * as o3 from "oasis-engine";
+import { OrbitControl } from "@oasis-engine/controls";
+import { MeshTopology } from "oasis-engine";
+
+const engine = new o3.WebGLEngine("o3-demo");
```
Comment (eyworldwide) [inline demos/model-mesh-obj-loader/index.ts:5]: Change canvas id to `canvas`

File: demos/model-mesh-obj-loader/index.ts (hunk 3 above)
Comment (gz65555) [reply demos/model-mesh-obj-loader/index.ts:44]: why? I'll no longer change the data.

**Comment by gz65555** [comment]
I reopen a pull request #83 to dev/0.4, which has used new engine API.

**Pull request closed without merging**